modules/cart/src/CartManager.php

```diff
@@ -110,6 +110,7 @@ public function addLineItem(OrderInterface $cart, LineItemInterface $line_item,
       $new_quantity = $matching_line_item->getQuantity() + $quantity;
       $matching_line_item->setQuantity($new_quantity);
       $matching_line_item->save();
+      $needs_cart_save = TRUE;
```

Comment:
scroll down a bit. you should see this just below the else:

    if ($needs_cart_save && $save_cart) {
      $cart->save();
    }